pkg/hostagent/setup.go

```diff
@@ -284,6 +292,21 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 			}
 		}
 
+		for _, ip := range iface.IPs {
+			// We are handling 4 cases  here: ipv4 only, ipv6 only, ipv6 and ipv4, and ipv4 and ipv6
+			if ip.Address.IP != nil {
+				if iface.HostVethName != "" && iface.Mac != "" {
```

Comment:
Nit: I don't think we really need this check

can't we simplify the block with
```
iface.HostVethName, iface.Mac, err =
    runSetupVeth(iface.Sandbox, iface.Name, mtu, ip.Address.IP)

if err != nil {
    return nil, err
} else {
    break
}
```

Reply:
Okay sure. With the above changes where we allocate mac address based on ipv4 only , and the rest remain same for Ipv6, this should handle both ipv4 only, ipv6 only, dual-stack with ipv6 first or idual stack Ipv4 first.